api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/DetermineEntitlementMessageProcessor.java

```diff
@@ -69,26 +72,48 @@ public MessageStatus processMessage(Message message) {
                 previousPaymentCycle);
 
         paymentCycleService.updatePaymentCycle(currentPaymentCycle, decision);
-        handleDecision(claim, currentPaymentCycle, decision);
+        handleDecision(claim, previousPaymentCycle, currentPaymentCycle, decision);
         return COMPLETED;
     }
 
-    private void handleDecision(Claim claim, PaymentCycle currentPaymentCycle, EligibilityAndEntitlementDecision decision) {
+    private void handleDecision(Claim claim, PaymentCycle previousPaymentCycle, PaymentCycle currentPaymentCycle, EligibilityAndEntitlementDecision decision) {
         if (decision.getEligibilityStatus() == ELIGIBLE) {
             createMakePaymentMessage(currentPaymentCycle);
-        } else if (decision.noChildrenPresentInCurrentCycle() && claimantIsNotPregnant(claim, currentPaymentCycle)) {
-            handleNoLongerEligibleForScheme(claim);
-        } else {
-            handleLossOfQualifyingBenefitStatus(claim);
+        } else if (claim.getClaimStatus() == ACTIVE) {
+            if (shouldExpireClaim(decision, previousPaymentCycle, currentPaymentCycle)) {
+                updateClaimStatus(claim, ClaimStatus.EXPIRED);
+            } else if (decision.getQualifyingBenefitEligibilityStatus().isNotEligible()) {
+                handleLossOfQualifyingBenefitStatus(claim);
+            } else {
+                handleNoLongerEligibleForSchemeAsNoChildrenAndNotPregnant(claim);
+            }
+        }
+        //TODO HTBHF-1296: PENDING_EXPIRY will be moved to EXPIRED after 16 weeks.
+        // else {
+        // }
+    }
+
+    private boolean shouldExpireClaim(EligibilityAndEntitlementDecision decision, PaymentCycle previousPaymentCycle, PaymentCycle currentPaymentCycle) {
+        if (decision.childrenPresent() || claimantIsPregnantInCycle(currentPaymentCycle)) {
+            return false;
         }
+        if (childrenExistedInPreviousCycleAndNowOver4(previousPaymentCycle, currentPaymentCycle)) {
+            return true;
+        }
+        return claimantIsPregnantInCycle(previousPaymentCycle) && !claimantIsPregnantInCycle(currentPaymentCycle);
```

Comment:
Do we need to check if the claimant was pregnant previously?
I think we're missing the case where the children go off the feed without turning four. (loss of child).

Reply:
That case is covered and tested. In that scenario we would fall through this method and the claim would not be expired, it would then either go to `handleLossOfQualifyingBenefitStatus` if they were not eligible with DWP or `handleNoLongerEligibleForSchemeAsNoChildrenAndNotPregnant` if they were still eligible.